modules/dnn/src/layers/fast_convolution/fast_convolution.simd.hpp

```diff
@@ -0,0 +1,346 @@
+
+        c21 = vfmaq_laneq_f32(c21, b0, a0, 3);
+
+        b0 = vld1q_f32(b);
+        b += 4;
+
+        c22 = vfmaq_laneq_f32(c22, b1, a0, 3);
+        c23 = vfmaq_laneq_f32(c23, b2, a0, 3);
+
+        c24 = vfmaq_laneq_f32(c24, b0, a0, 0);
+        c25 = vfmaq_laneq_f32(c25, b0, a0, 1);
+        c26 = vfmaq_laneq_f32(c26, b0, a0, 2);
+        c27 = vfmaq_laneq_f32(c27, b0, a0, 3);
+    }
+
+    if (ifActiv)
+    {
+        b0 = vdupq_n_f32(minval), b1 = vdupq_n_f32(maxval);
+        c0 = vminq_f32(vmaxq_f32(c0, b0), b1);
+        c1 = vminq_f32(vmaxq_f32(c1, b0), b1);
+        c2 = vminq_f32(vmaxq_f32(c2, b0), b1);
+        c3 = vminq_f32(vmaxq_f32(c3, b0), b1);
+        c4 = vminq_f32(vmaxq_f32(c4, b0), b1);
+        c5 = vminq_f32(vmaxq_f32(c5, b0), b1);
+        c6 = vminq_f32(vmaxq_f32(c6, b0), b1);
+        c7 = vminq_f32(vmaxq_f32(c7, b0), b1);
+        c8 = vminq_f32(vmaxq_f32(c8, b0), b1);
+        c9 = vminq_f32(vmaxq_f32(c9, b0), b1);
+        c10 = vminq_f32(vmaxq_f32(c10, b0), b1);
+        c11 = vminq_f32(vmaxq_f32(c11, b0), b1);
+        c12 = vminq_f32(vmaxq_f32(c12, b0), b1);
+        c13 = vminq_f32(vmaxq_f32(c13, b0), b1);
+        c14 = vminq_f32(vmaxq_f32(c14, b0), b1);
+        c15 = vminq_f32(vmaxq_f32(c15, b0), b1);
+        c16 = vminq_f32(vmaxq_f32(c16, b0), b1);
+        c17 = vminq_f32(vmaxq_f32(c17, b0), b1);
+        c18 = vminq_f32(vmaxq_f32(c18, b0), b1);
+        c19 = vminq_f32(vmaxq_f32(c19, b0), b1);
+        c20 = vminq_f32(vmaxq_f32(c20, b0), b1);
+        c21 = vminq_f32(vmaxq_f32(c21, b0), b1);
+        c22 = vminq_f32(vmaxq_f32(c22, b0), b1);
+        c23 = vminq_f32(vmaxq_f32(c23, b0), b1);
+        c24 = vminq_f32(vmaxq_f32(c24, b0), b1);
+        c25 = vminq_f32(vmaxq_f32(c25, b0), b1);
+        c26 = vminq_f32(vmaxq_f32(c26, b0), b1);
+        c27 = vminq_f32(vmaxq_f32(c27, b0), b1);
+    }
+    vst1q_f32(c, c0);
+    vst1q_f32(c + 4, c1);
+    vst1q_f32(c + 8, c2);
+    vst1q_f32(c + 12, c3);
+    vst1q_f32(c + 16, c4);
+    vst1q_f32(c + 20, c5);
+    vst1q_f32(c + 24, c24);
+
+    vst1q_f32(c + ldc, c6);
+    vst1q_f32(c + ldc + 4, c7);
+    vst1q_f32(c + ldc + 8, c8);
+    vst1q_f32(c + ldc + 12, c9);
+    vst1q_f32(c + ldc + 16, c10);
+    vst1q_f32(c + ldc + 20, c11);
+    vst1q_f32(c + ldc + 24, c25);
+
+    vst1q_f32(c + ldc * 2, c12);
+    vst1q_f32(c + ldc * 2 + 4, c13);
+    vst1q_f32(c + ldc * 2 + 8, c14);
+    vst1q_f32(c + ldc * 2 + 12, c15);
+    vst1q_f32(c + ldc * 2 + 16, c16);
+    vst1q_f32(c + ldc * 2 + 20, c17);
+    vst1q_f32(c + ldc * 2 + 24, c26);
+
+    vst1q_f32(c + ldc * 3, c18);
+    vst1q_f32(c + ldc * 3 + 4, c19);
+    vst1q_f32(c + ldc * 3 + 8, c20);
+    vst1q_f32(c + ldc * 3 + 12, c21);
+    vst1q_f32(c + ldc * 3 + 16, c22);
+    vst1q_f32(c + ldc * 3 + 20, c23);
+    vst1q_f32(c + ldc * 3 + 24, c27);
+#else
+    CV_LOG_WARNING(NULL, "Runing at unoptimized code. The combination of FAST_CONV_MR and FAST_CONV_NR is not supported in NEON branch.");
```

Comment:
I think, it should be compile-time error in this case.
```
#error "unsupported FAST_CONV_MR and/or FAST_CONV_NR"
```

Reply:
fixed.